samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -74,7 +74,7 @@ public GroupByContainerCount(int containerCount) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(i, taskGroups[i]));
+      containerModels.add(new ContainerModel(String.valueOf(i), taskGroups[i]));
```

Comment:
> I suggest that we at least add a validation with a clear exception that ensures that the external strings (e.g. those that come from the TaskAssignmentManager) are all parseable integers.

What kind of validation would you suggest? In the Grouper itself? 

> A better long term approach is to completely overhaul these groupers to have no assumptions about the strings and treat them as arbitrary tokens

Yes. That was the intention with the standalone work. The task to replace with the new grouper has not yet been completed. I will add a TODO with a JIRA here so that we don't forget about it. For existing users running Yarn, this should continue to work as long as the AppMaster logic is not changed.